wp-includes/taxonomy.php

```diff
@@ -108,25 +112,92 @@ function create_initial_taxonomies() {
 		)
 	);
 
+	// Check the current theme supports Post Formats.
+	$post_format_support = current_theme_supports( 'post-formats' );
+
 	register_taxonomy(
 		'post_format',
 		'post',
 		array(
 			'public'            => true,
 			'hierarchical'      => false,
 			'labels'            => array(
-				'name'          => _x( 'Formats', 'post format' ),
-				'singular_name' => _x( 'Format', 'post format' ),
+				'name'                  => _x( 'Formats', 'post format' ),
+				'singular_name'         => _x( 'Format', 'post format' ),
+				'back_to_items'         => _x( '&larr; Go to Formats', 'post format' ),
+				'edit_item'             => _x( 'Edit Format', 'post format' ),
+				'item_link'             => _x( 'Format Link', 'post format' ),
+				'item_link_description' => _x( 'A link to a post format.', 'post format' ),
+				'items_list'            => _x( 'Formats list', 'post format' ),
+				'items_list_navigation' => _x( 'Formats list navigation', 'post format' ),
+				'no_terms'              => _x( 'No formats', 'post format' ),
+				'not_found'             => _x( 'No formats found.', 'post format' ),
+				'search_items'          => _x( 'Search Formats', 'post format' ),
+				'update_item'           => _x( 'Update Format', 'post format' ),
+				'view_item'             => _x( 'View Format', 'post format' ),
 			),
 			'query_var'         => true,
 			'rewrite'           => $rewrite['post_format'],
-			'show_ui'           => false,
+			'show_ui'           => $post_format_support,
 			'_builtin'          => true,
-			'show_in_nav_menus' => current_theme_supports( 'post-formats' ),
+			'capabilities'      => array(
+				'create_terms' => 'do_not_allow',
+				'manage_terms' => 'manage_post_formats',
+				'edit_terms'   => 'edit_post_formats',
+				'delete_terms' => 'do_not_allow',
+				'assign_terms' => 'assign_post_formats',
+			),
+			'show_in_nav_menus' => $post_format_support,
 			'show_in_rest'      => true,
 		)
 	);
 
+	/*
+	 * As the Post Format feature is using the term slug as an ID, using a term meta
+	 * will help Site Admins customize the Post Format slugs to match their native
+	 * language.
+	 */
+	register_term_meta(
+		'post_format',
+		'post_format_slug',
```

Comment:
```suggestion
		'post_format_custom_slug',
```
Let's stress the fact it's a custom slug.